blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/htmlpublisher/HTMLArtifact.java

```diff
@@ -0,0 +1,64 @@
+package io.jenkins.blueocean.htmlpublisher;
+
+import com.google.common.collect.Lists;
+import htmlpublisher.HtmlPublisherTarget;
+import hudson.Extension;
+import hudson.model.Run;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.factory.BlueArtifactFactory;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BlueArtifact;
+
+import java.util.Collection;
+import java.util.List;
+
+public class HTMLArtifact extends BlueArtifact {
+
+    private final HtmlPublisherTarget.HTMLBuildAction action;
+
+    public HTMLArtifact(HtmlPublisherTarget.HTMLBuildAction action, Link parent) {
+        super(parent);
+        this.action = action;
+    }
+
+    @Override
+    public String getName() {
+        return action.getHTMLTarget().getReportName();
+    }
+
+    @Override
+    public String getPath() {
+        return action.getHTMLTarget().getReportName();
+    }
+
+    @Override
+    public String getUrl() {
+        return String.format("/%s%s", action.getOwner().getUrl(), action.getUrlName());
```

Comment:
Thats the URL for the action as constructed by the HTMLPublish plugin. Out of my control.